stout/range.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <algorithm>
+
+#include "stout/stream.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+namespace detail {
+
+////////////////////////////////////////////////////////////////////////
+
+struct _Range {
+  template <typename K_, typename Arg_>
+  struct Continuation : public TypeErasedStream {
+    // NOTE: explicit constructor because inheriting 'TypeErasedStream'.
+    Continuation(K_ k, int from, int to)
+      : k_(std::move(k)),
+        to_(to),
+        from_(std::min(from, to)) {}
+
+    void Start() {
+      k_.Start(*this);
+    }
+
+    template <typename... Args>
+    void Fail(Args&&... args) {
+      k_.Fail(std::forward<Args>(args)...);
+    }
+
+    void Stop() {
+      k_.Stop();
+    }
+
+    void Ended() {
+      k_.Ended();
+    }
+
+    void Next() override {
+      if (from_ == to_) {
+        k_.Ended();
+      } else {
+        k_.Body(from_++);
+      }
+    }
+
+    void Done() override {
+      k_.Ended();
+    }
+
+    K_ k_;
+    int to_;
+    int from_;
+  };
+
+  template <typename T>
```

Comment:
Since you're not using this functionality right now (you're always using 'int') and it's really trivial to add later let's drop the template here.